### PR TITLE
sys-kernel/coreos-*: add overlay ebusy fix

### PR DESCRIPTION
**Pull request by euank**
Goes along with https://github.com/coreos/linux/pull/96

### CODE REVIEW

**Comment by euank** [comment]
Added that change too

**Comment by euank** [comment]
Thanks for the reminder. Revert PR: https://github.com/coreos/docker/pull/43

**Comment by euank** [comment]
This built + worked locally. Hopefully I didn't miss anything else.

File: app-emulation/docker/docker-9999.ebuild [hunk 1]
```diff
@@ -212,7 +210,7 @@ src_unpack() {
 		DOCKER_GITCOMMIT=$(git -C "${S}" rev-parse HEAD | head -c 7)
 		DOCKER_BUILD_DATE=$(git -C "${S}" log -1 --format="%ct")
 	fi
-	eapply -d"${S}"/components/engine "${ENGINE_PATCHES[@]}"
+	[ -n "${ENGINE_PATCHES}" ] && eapply -d"${S}"/components/engine "${ENGINE_PATCHES[@]}"
```
Comment (dm0-) [inline app-emulation/docker/docker-9999.ebuild]: It's minor, but I think I'd rather this test for a non-empty string using the `[*]` subscript, or just test the length of the array.  As is, it references the non-existent first array element (which is an error with `-u`).

**Pull request merged**